### PR TITLE
feat: quality of life improvemnet

### PR DESCRIPTION
**Pull request by Eastonco**
# Description
* added new pretty printing
* updated config.json layout
* added a new example config file
* added setup instructions
* fixed outdated code
* refactored bits and bobs

# Screenshots
Screen Shot 2023-06-26 at 10 53 34 PM

# Checklist
- [x] My code has been tested locally
- [x] documentation has been updated accordingly
- [ ]

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -70,23 +64,24 @@ def browser_setup():
...
+    delta = afterPageLoad - beforePageLoad
+    printInfo(afterPageLoad.strftime("%H:%M:%S"),
+              " - Page loaded, took ", delta.seconds, 's')
+
     free_elements = browser.find_elements(
         by=By.CLASS_NAME, value='cl-search-result')
     for el in free_elements:
-
-        a_tag = el.find_element(by=By.CLASS_NAME, value='titlestring')
+        a_tag = el.find_element(By.CLASS_NAME, 'posting-title')
```
Comment (Eastonco) [inline main.py:84]: This anchor tag had changed and was causing failures

File: main.py [hunk 2]
```diff
@@ -58,10 +54,8 @@
 
 
 def browser_setup():
-    user_agent = 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/109.0'
     firefox_option = Options()
-    firefox_option.add_argument('-headless')
-    firefox_option.set_preference('general.useragent.override', user_agent)
+    firefox_option.add_argument('--headless')
```
Comment (Eastonco) [inline main.py:58]: headless flag broken on my instance, needed the `--` to work. Also remove the user_agent as it didn't seem needed

File: main.py [hunk 3]
```diff
@@ -112,8 +107,6 @@ def scrape(url: str, timestamp: str):
         except Exception as exc:
             browser.quit()
             error_count += 1
-            print(
-                f'ERROR: {exc=}\nScript will try {3-error_count} more times and then shutdown.')
```
Comment (Eastonco) [inline main.py:116]: moved the error logging into the send error alert function

File: main.py [hunk 4]
```diff
@@ -124,84 +117,80 @@ def scrape(url: str, timestamp: str):
...
 
     return num_listings
 
 
-def welcome_message():
-    print(f'''
-    Welcome to the craigslist free alert searcher. 
-    Every 3 minutes this script will query cl for free items and then print the new items to the terminal.
-    cheers!
-    ''')
```
Comment (Eastonco) [inline main.py:150]: moved to `prettyPrint.py`

File: main.py [hunk 5]
```diff
@@ -124,84 +117,80 @@ def scrape(url: str, timestamp: str):
...
+    msg['From'] = config.email.src
+    msg['To'] = config.emails.dst
+    message_body = f'title: {alert.title}\nscraped: {alert.time_scraped}\nposted: {alert.time_posted}\nlocation:{alert.location}\n{alert.link}'
+    msg.set_content(message_body)
 
     ssl_context = ssl.create_default_context()
     with smtplib.SMTP_SSL("smtp.gmail.com", 465, context=ssl_context) as server:
-        server.login(config.src_email, config.email_key)
+        server.login(config.email.src, config.email.email_key)
         server.send_message(msg)
```
Comment (Eastonco) [inline main.py]: wasn't able to verify if this works but `alert.to_json()` was throwing errors

**Review by Eastonco — COMMENTED** [review]
lmk what you think!

**Comment by silasstokes** [comment]
This looks awesome. I didn't realize there was so much cleaning up that needed to be done - I owe you one for going the extra mile and getting this ship sea worthy again. 

I am going to go ahead and merge it!

**Pull request merged**